crates/namada/src/vm/wasm/run.rs

```diff
@@ -2117,4 +2195,22 @@ mod tests {
                 Either::Left(wasmer_vm::TrapCode::StackOverflow),
         );
     }
+
+    /// The following definitions are copied from wasmer v4.3.5
+    /// `lib/compiler/src/engine/trap/frame_info.rs` to access internal
+    /// fields that are otherwise private. This must be carefully maintained
+    /// while we workaround the leak before it's fixed in wasmer.
+    pub struct GlobalFrameInfo {
+        ranges: BTreeMap<usize, ModuleInfoFrameInfo>,
+    }
+    struct ModuleInfoFrameInfo {
+        _start: usize,
+        _functions: BTreeMap<usize, FunctionInfo>,
+        _module: std::sync::Arc<wasmer_types::ModuleInfo>,
+        _frame_infos: wasmer_compiler::FrameInfosVariant,
+    }
+    struct FunctionInfo {
+        _start: usize,
+        _local_index: wasmer_types::LocalFunctionIndex,
+    }
```

Comment:
I'm fine with this hack, but be mindful that these aren't part of the public API. in cargo, we set the `wasmer` version to `4.3.5`, however, I think the underlying version may change depending on the interaction with other dependencies. maybe this isn't the case for us since we commit the cargo lock file... anyway, the point is, upstream is free to change the structure of `GlobalFrameInfo`

Reply:
dully noted in docstring above :) I think there's no danger in a test - if the api changes it's only going to blow up in a test